### PR TITLE
Rails Budget App

### PR DESCRIPTION
**Pull request by principles31**
# Ruby on Rails capstone project - Budget app:
> The aim of this project is about building a mobile web application where you can manage your budget: you have a list of transactions associated with a category so that you can see how much money you spent and on what.

- You will create a Ruby on Rails application that allows the user to:

> register and log in, so that the data is private to them.
> introduce new transactions associated with a category.
> see the money spent on each category.

## The following were implemented as the project requirements:
- Followed these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=),
- Thanks to The design template provider [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).
> Interactions
  - Splash screen

- A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
> Sign up and log in pages

 - The user should be able to register in the app with full name, email and password (all mandatory).
 - The user can log into the app using email and password.
 - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
> Home page (catagories page)

 - When the user logs in, they are presented with the categories page.
 - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that > >
 > category.
 - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
 - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
> Transactions page

 - For a given category, the list of transactions is presented, ordered by the most recent.
 - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
 - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
 - When the user clicks on the "Back" button (<), the user navigates to the home page.
> "Add a new category" page

 - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
 - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
 - When the user clicks on the "Back" button (<), the user navigates to the home page.
> "Add a new transaction" page

 - The user fills out a form to create a new transaction with:
 - name (mandatory)
 - amount (mandatory)
> categories (mandatory at least one)
 - The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
 - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
> Testing requirements
 - Create unit and integration tests for all the most important components of your RoR application.
> Technical requirements
 - Used Postgres as your database.
 - Used devise for authentication.
 - Validate all user input to make sure that anyone with bad intentions cannot compromise your app.
 - You can use a view template engine of your choice (.erb, .slim, .haml).
 - The project should be deployed and accessible online.
 - Your database schema should reflect the following structure:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,73 @@
-# Rails_Budget_App
\ No newline at end of file
+# Rails_Budget_App
+# ![](https://img.shields.io/badge/Microverse-blueviolet) 
+
+## [Live Version]
```
Comment (VICTOR-CHUKWU) [inline README.md]: - [ ] Nice job @principles31 , I can not see the live link of your project, kindly include that

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @principles31 , 
## STATUS: REQUIRED CHANGES :recycle: 
Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
- [ ] KIndly include the config.ru file in your root directory so that anyone who clones the project can run it with `rails s`

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 1 above)
Comment (principles31) [reply README.md]: @VICTOR-CHUKWU Thank you for the feedback

**Comment by principles31** [comment]
- The changes requested by the previous reviewer have been implemented.

File: README.md (hunk 1 above)
Comment (muneebulrehman) [reply README.md]: Hey, the link you provided seems to be broken. Kindly add a valid link so the next reviewer can check the deployed version of your application. 
image

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @principles31 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Highlights
- No linter errors
- Descriptive PR

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] As requested by the previous reviewer, kindly add the `config.ru` file to your root directory, so that if some one clones your repo, they can run the `rails s` or `rails server` command and start the development server.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by principles31** [comment]
- [x] As requested by the previous reviewer, kindly add the config.ru file to your root directory, so that if some one clones your repo, they can run the rails s or rails server command and start the development server.
- [ ] The above change have been implemented

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @principles31, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by principles31** [comment]
@sja-thedude Thank you for your approval.

**Pull request merged**